packages/mds-policy-service/repository/index.ts

```diff
@@ -14,14 +14,336 @@
  * limitations under the License.
  */
 
-import { ReadWriteRepository } from '@mds-core/mds-repository'
+import db from '@mds-core/mds-db'
+import { InsertReturning, ReadWriteRepository, RepositoryError } from '@mds-core/mds-repository'
+import { Timestamp, UUID } from '@mds-core/mds-types'
+import {
+  AlreadyPublishedError,
+  BadParamsError,
+  ConflictError,
+  DependencyMissingError,
+  NotFoundError,
+  now
+} from '@mds-core/mds-utils'
+import { PolicyDomainModel, PolicyMetadataDomainModel, ReadPolicyQueryParams } from '../@types'
 import entities from './entities'
+import { PolicyEntity } from './entities/policy-entity'
+import { PolicyMetadataEntity } from './entities/policy-metadata-entity'
+import {
+  PolicyDomainToEntityCreate,
+  PolicyEntityToDomain,
+  PolicyMetadataDomainToEntityCreate,
+  PolicyMetadataEntityToDomain
+} from './mappers'
 import migrations from './migrations'
-
 class PolicyReadWriteRepository extends ReadWriteRepository {
   constructor() {
     super('policies', { entities, migrations })
   }
+
+  public readPolicies = async (params?: ReadPolicyQueryParams) => {
+    try {
+      const connection = await this.connect('ro')
+      const query = connection.getRepository(PolicyEntity).createQueryBuilder()
+
+      if (params) {
+        const { policy_ids, rule_id, get_unpublished, get_published, start_date, geography_id } = params
+
+        if (policy_ids) {
+          query.andWhere('policy_id = ANY(:policy_ids)', { policy_ids })
+        }
+
+        if (rule_id) {
+          query.andWhere(
+            "EXISTS(SELECT FROM json_array_elements(policy_json->'rules') elem WHERE (elem->'rule_id')::jsonb ? :rule_id)",
+            { rule_id }
+          )
+        }
+
+        if (get_unpublished) {
+          query.andWhere("policy_json->>'publish_date' IS NULL")
+        }
+
+        if (get_published) {
+          query.andWhere("policy_json->>'publish_date' IS NOT NULL")
+        }
+
+        if (get_unpublished && get_published) {
+          throw new BadParamsError('cannot have get_unpublished and get_published both be true')
+        }
+
+        if (start_date) {
+          query.andWhere("policy_json->>'start_date' >= '${params.start_date}'", { start_date })
+        }
+
+        if (geography_id) {
+          query.andWhere(`policy_json::jsonb @> :json_query`, {
+            json_query: { rules: [{ geographies: [geography_id] }] }
+          })
+        }
+      }
+
+      const entities = await query.getMany()
+      return entities.map(PolicyEntityToDomain.map)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  public readActivePolicies = async (timestamp: Timestamp = now()) => {
+    try {
+      const connection = await this.connect('ro')
+      const entities = await connection
+        .getRepository(PolicyEntity)
+        .createQueryBuilder()
+        .where("policy_json->>'start_date' <= :start_date", { start_date: timestamp })
```

Comment:
Should be `"policy_json->>'start_date' >= :start_date"`.

Reply:
thanks!